File: Java/Inheritance/SubclassTest/payroll/SalesPerson.java
package payroll;

//subclass SalesPerson inherits from superclass Employee
public class SalesPerson extends Employee {

	private double sales;

	public SalesPerson(int h, float r, double s) {
		super(h, r); //calling superclass constructor
		sales = s;
	}

	public double getSales() { return sales; }

	//overriding method of superclass
	public double getIncome() {
		double income = super.getIncome();
		if(sales >= 20000)
			income += 0.05 * sales;
		return income;
	}
}

File: DotNet/Inheritance/DerivedClassTest/DemoApp/Payroll/SalesPerson.cs
namespace Payroll
{
    //SalesPerson class is derived from Employee (base) class
    class SalesPerson : Employee
    {
        //automatic property
        public double Sales {get; set;}

        public SalesPerson(int h, float r, double s) : base(h, r)
        {
            Sales = s;
        }

        public override double Income()
        {
            double income = base.Income();
            if(Sales >= 20000)
                income += 0.05 * Sales;
            return income;
        }
    }
}